pinot-core/src/main/java/com/linkedin/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManager.java

```diff
@@ -252,6 +256,19 @@ protected void consumeLoop() {
       } catch (TimeoutException e) {
         segmentLogger.warn("Timed out when fetching messages from Kafka, retrying");
         continue;
+      } catch (SimpleConsumerWrapper.TransientConsumerException e) {
+        consecutiveErrorCount++;
+
+        if (consecutiveErrorCount < MAX_CONSECUTIVE_ERROR_COUNT) {
+          successful = false;
+          break;
```

Comment:
Please rethrow the exception here, so that the state machine goes into an ERROR state (for now). We will handle the specific exceptions later.